BundleMon seems to fail intermittently

Here's a recent commit where `BundleMon` job fails: https://github.com/Leaflet/Leaflet/runs/4930516600?check_suite_focus=true

```
Error:  create commit record: bad project credentials
{"error":"GitHub action 1742977939 should be \"in_progress\" status"}
Error:  Failed to create commit record
Error:  Unhandled error
Error: Failed to generate report
    at /home/runner/work/Leaflet/Leaflet/node_modules/bundlemon/lib/main/index.js:32:15
    at Generator.next (<anonymous>)
    at fulfilled (/home/runner/work/Leaflet/Leaflet/node_modules/bundlemon/lib/main/index.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Error: Process completed with exit code 1.
```

Not sure what this means but probably worth investigating, or if it's expected, disabling the job if it can't run.

Looks like it happens sporadically and for maintainer commits as well, not just contributors. Here's the latest occurrence: https://github.com/Leaflet/Leaflet/runs/4964248089?check_suite_focus=true

```
Error:  create commit record: bad project credentials
{"error":"GitHub action status should be \"in_progress\""}
Error:  Failed to create commit record
Error:  Unhandled error
Error: Failed to generate report
    at /home/runner/work/Leaflet/Leaflet/node_modules/bundlemon/lib/main/index.js:32:15
    at Generator.next (<anonymous>)
    at fulfilled (/home/runner/work/Leaflet/Leaflet/node_modules/bundlemon/lib/main/index.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -111,6 +111,7 @@ jobs:
 
   bundlemon:
     needs: build-release
+    if: github.repository_owner == 'Leaflet'
     runs-on: ubuntu-latest
     steps:
       - name: Restore release build

